acp/acp.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package acp
+
+import (
+	"context"
+
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/corelog"
+)
+
+var (
+	log = corelog.NewLogger("acp")
+
+	// NoACPModule is an empty ACPModule, this is used to disable access control.
+	NoACPModule = immutable.None[ACPModule]()
+)
+
+// ACPModule is the interface to all types of access control modules that might exist.
+type ACPModule interface {
```

Comment:
nitpick/suggestion: I'm not a fan of the `Module` suffix here, as everything is some what a "module". I assume this was inherited by the fact that we refer to the SourceHub ACP implementation as the ACP Module, but that is language that is specific to the CosmosSDK in how you implement/extend functionality within the SDK, as *everything* is organized as "modules" (staking module, bank module, acp module, etc).

But I dont think this langauge is necessary here. The applies broadly to all instances of "Module" term for funcs/types in this PR (eg: `NoACPModule, `WithACPModule`, etc..)

Reply:
`acp.ACP` is perfectly fine type name ;)

Reply:
I like module (even though it might have trickled down from CosmosSDK). It refers to that "thing" that contains the acp logic or system / or model (whatever you want to think of it as). ACP is what this module provides (it also has it's own datastore logic separate from the database).

Reply:
Keeping as is, unless consensus or a harder preference.

Reply:
why not `PlannerModule` or `NetModule` or `ParserModule`? These are the "things" the contain "planner", "net", and "parsing", no?

`net` has its own encapsulated logic, even its own datastore off the root.

I can't see a reason for the naming other than leaking from the terminology from the Protocol team. If it was consistent with anything else in the DB, I could get behind it.

Reply:
I don't mind seeing `NetModule` in that case if it was :P

How hard of a preference do you have for this? I see this marked as a `nitpick/suggestion`.

Reply:
I will change the Terminology from ACPModule -> ACP 

- [ ] Change ACPModule -> ACP

Reply:
Done in commit: ``` PR(ACP): Remove Module Terminology From ACP ```